main.py

```diff
@@ -52,13 +68,69 @@ def configure_confluent_kafka_consumer(event, args):
     if logger.isEnabledFor(logging.DEBUG):
         # Log everything from boto3
         boto3.set_stream_logger()
-        logger.debug(f"Using boto3 {boto3.__version__}")
+    logger.debug(f"Using boto3 {boto3.__version__}")
 
     logger.debug(event)
 
     message = json.loads(event['Records'][0]['Sns']['Message'])
     logger.debug(message)
 
+    private_ip = message['detail']['containers'][0]['networkInterfaces'][0]['privateIpv4Address']
+    logger.debug(private_ip)
+
+    connector_config = {
+        "connector.class": "io.confluent.connect.s3.S3SinkConnector",
+        "tasks.max": args.tasks_max,
+        "topics": args.topics,
+        "flush.size": args.flush_size,
+        "s3.region": args.aws_region,
+        "s3.bucket.name": args.s3_bucket_name,
+        "storage.class": "io.confluent.connect.s3.storage.S3Storage",
+        "format.class": "io.confluent.connect.s3.format.avro.AvroFormat",
+        "schema.generator.class": "io.confluent.connect.storage.hive.schema.DefaultSchemaGenerator",
+        "partitioner.class": "io.confluent.connect.storage.partitioner.DefaultPartitioner",
+    }
+    payload = {
+        "name": args.connector_name,
+        "config": connector_config
+    }
+
+    # Get a list of existing connectors
+
+    response = requests.get(f"http://{private_ip}:{args.port}/connectors")
+    existing_connectors = response.content.decode("utf-8")
+    logger.debug(existing_connectors)
+
+    # PUT payload if connectors do exist this updates existing connector
+
+    if args.connector_name in existing_connectors:
+        logger.debug("update connector [PUT]")
+
+        payload = {
+            "config": connector_config
+        }
+
+        response = requests.put(f"http://{private_ip}:{args.port}/connectors/{args.connector_name}/config",
+                                json=payload)
+        logger.debug(response.content.decode("utf-8"))
+
+    else:
+        # POST payload if connectors don't exist
+
+        logger.debug("create connector [POST]")
+
+        response = requests.post(f"http://{private_ip}:{args.port}/connectors", json=payload)
+        logger.debug(response.content.decode("utf-8"))
+
+        # DELETE all others
+
+    for existing_connector in existing_connectors:
```

Comment:
there is a problem with the logic here. How it currently is, it will delete _all_ connectors, including the one we just created/updated.

you need a: `if existing_connector is not args.connector_name:` below this line and indent  the other 3 lines under it